### PR TITLE
A smörgåsbord of lint/build tweaks

### PR DESCRIPTION
**Pull request by daviddrysdale**
Best reviewed commit-by-commit.

### CODE REVIEW

**Comment by daviddrysdale** [comment]
(A smörgåsbord because we can pick and choose which bits to include...)

File: scripts/build_server_docker [hunk 1]
```diff
@@ -19,13 +19,21 @@ fi
 if [[ -f "$OAK_REMOTE_CACHE_KEY" ]]; then
     "$SCRIPTS_DIR/docker_run" bazel build \
         ${BAZEL_OPTS:+ "${BAZEL_OPTS}"} \
+        --incompatible_disable_deprecated_attr_params=false \
```
Comment (michael-kernel-sanders) [inline scripts/build_server_docker]: Add these to a variable?

Comment (anghelcovici) [reply scripts/build_server_docker]: Can you also add some comments of why we need this? (and maybe we should have GitHub issue to remove them when we upgrade to the correct thing)

File: scripts/build_server_local [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
```
Comment (anghelcovici) [inline scripts/build_server_local]: I think maintaining both a local and a docker build is going to be difficult. We should probably decide on one or the other. Let's have a chat about pros and cons when @tiziano88 is back

File: scripts/check_formatting [hunk 3]
```diff
@@ -4,29 +4,13 @@ set -o errexit
...
-  && CLANG_FORMAT_EXIT_CODE=$? || CLANG_FORMAT_EXIT_CODE=$?
-
-# When looking for clang-format "replacement" tags, grep will return 0 if it
-# has found anything and 1 if not, so here we'll invert the return code to
-# indicate an error when the formatting is insufficient.
-if [ $CLANG_FORMAT_EXIT_CODE -ne 1 ]; then
-  echo 'Please format code with clang-format'
-  exit 1
-fi
+./third_party/run-clang-format/run-clang-format.py -r oak examples
```
Comment (anghelcovici) [inline scripts/check_formatting]: What is this giving us on top of what we have?

File: scripts/build_server_local (hunk 2 above)
Comment (daviddrysdale) [reply scripts/build_server_local]: I'll drop this commit from the PR and we can discuss later.  (I've also realized it needs more tweaking anyhow &ndash; switching between client and server builds invalidates the Bazel cached outputs, as they use different toolchains.)

However, I will note that the speed difference between a local build and a Docker build is quite large (60x)&hellip;
```console
# Second run with no changes in between
% time ./scripts/build_server_docker  > /dev/null 2>&1
real	0m12.643s
user	0m0.103s
sys	0m0.096s
% time ./scripts/build_server_local > /dev/null 2>&1
real	0m0.213s
user	0m0.052s
sys	0m0.026s
```

File: scripts/build_server_docker (hunk 1 above)
Comment (daviddrysdale) [reply scripts/build_server_docker]: Done, with an inline `TODO` and more information in the commit message.

**Comment by daviddrysdale** [comment]
For the question of tooling in something other than shell scripts, I've tried that with Python in the past and it ended up being more unwieldy than just using a shell.  But maybe Rust has better libraries for process control, signalling, output redirection, grepping, text substitutions etc.

**Pull request merged**